app/controllers/v1/users/registrations_controller.rb

```diff
@@ -0,0 +1,32 @@
+module V1::Users
+  class RegistrationsController < ApplicationController
+    before_action :authenticate_user!, except: :create
+
+    def create
+      user = User.new(create_params)
+
+      if user.save
+        @current_user = user
```

Comment:
Z racji, że klasa kontrolera `RegistrationsController` jest umiejscowiona na końcu hierarchii dziedziczenia, ustawienie zmiennej `@current_user` w tym miejscu ma sens jedynie w sytuacji, gdy korzystamy ze zmiennej w obrębie tej klasy. W tym przypadku zmienna nie ma zastosowania. 

Moim zdaniem ustawienie `@current_user` możemy spokojnie ograniczyć do metody `authenticate_user!` w kontrolerze `ApplicationController`. Nastąpi to w kolejnym zapytaniu do API przy pomocy tokena otrzymanego w tej akcji kontrolera.